staging/src/k8s.io/client-go/openapi/groupversion.go

```diff
@@ -38,9 +39,23 @@ func newGroupVersion(client *client, item handler3.OpenAPIV3DiscoveryGroupVersio
 }
 
 func (g *groupversion) Schema(contentType string) ([]byte, error) {
-	return g.client.restClient.Get().
-		RequestURI(g.item.ServerRelativeURL).
-		SetHeader("Accept", contentType).
-		Do(context.TODO()).
-		Raw()
+	locator, err := url.Parse(g.item.ServerRelativeURL)
```

Comment:
Hmm… this assumes the serverRelativeURL returned is not actually a server relative URL, which is pretty weird

Reply:
This is the definition of server relative URL https://github.com/kubernetes/kube-openapi/blob/f5883ff37f0c685e2c0ad657c26c52baba49be6d/pkg/handler3/handler.go#L57-L58. `Parse` function actually parses relative URL seems not weird to me. But maybe I'm missing something?

Reply:
I mean we're parsing a "server relative URL" here and then using the path below in a way that is not actually server relative (we're appending it to an existing root path)

Reply:
ah, I see what you mean now. Although I agree that this might be counter-intuitive but this is the issue we're trying to solve, otherwise openapi v3 does not work on the endpoints including prefixes as in the referenced issue. And we can not change the name of `ServerRelativeURL` to better reflect. 
Do you have any suggestions that hasn't come to my mind?

Reply:
Does the server have visibility to the fact that it's being served with an added prefix? If so, can the server return server relative URLs that include the prefix? If not, can the thing proxying the server add in the prefix?

Reply:
Today, what the server returns is correct for standard server configurations, and incorrect for proxied+prefixed configurations.

Making the client assume the server returns incorrect information and unconditionally add in its own prefix limits what we can do on the server side to improve this in the future.

I would probably make this change conditional on `ServerRelativeURL` starting with `/openapi/v3/`. If `ServerRelativeURL` starts with something else, then treat it as an actual server-relative URL and continue using `RequestURI`. That would let us improve things on the server side in the future and not break clients containing this fix.

Reply:
Thanks a lot for your suggestion @liggitt and sounds good to me. 

I have one last question before making changes about adding a conditional of `ServerRelativeURL.startsWith(/openapi/v3)`;

openapi package was added only for openapiv3, would it be possible that `ServerRelativeURL` is different than `/openapi/v3`(or in the case that maybe in the far future, it will also serve `openapi/v4`)?

Reply:
@liggitt I updated PR. Could you PTAL?